winml/lib/Api/LearningModelBinding.cpp

```diff
@@ -611,7 +626,8 @@ void LearningModelBinding::BindUnboundOutputs() {
   // Add all unbound outputs to binding collection
   for (const auto& unbound_output : unbound_output_names) {
     Ort::Value out(nullptr);
-    WINML_THROW_IF_FAILED(BindOutput(unbound_output, out));
+    Ort::Allocator out_allocator(adapter_.get(), nullptr);
+    WINML_THROW_IF_FAILED(BindOutput(unbound_output, out, out_allocator));
```

Comment:
A few comments:

* BindOutput and BindInput are calling release() on the allocator. Does this loop work for multiple outputs? On the second iteration you're passing a null allocator, right?
  * If you didn't want BindOutput to modify the allocator, you can make the argument a `const Ort::Allocator&` so the compiler enforces that you're not changing it
* Why cache the adapter in Ort::Allocator and not use it? Every time you wans an allocator, you're releasing the old one and calling adapter_.get(). You could use move semantics instead since you created nice definitions for `Allocator(Allocator&& v)` and `void operator=(Allocator&& v)` above. The code will probably be prettier and you won't have to call adapter_.get() every time you want an allocator.